### ISSUE TITLE
support multiple configurations per regression test

### ISSUE THREAD

**Issue by vogler** [goblint/analyzer#243]
We should have a way to check the same regression test with multiple configurations to avoid duplicating files with just different `PARAM: ...` lines at the top.
Would be good to run several solvers on the same test. E.g. https://github.com/goblint/analyzer/blob/master/tests/regression/34-localization/01-nested.c should now also be tested with `td3` after https://github.com/goblint/analyzer/commit/cf11c4b0b3602f417a5573e56853b81940384998.
Also see https://github.com/goblint/analyzer/pull/196#issuecomment-850299611.

We could just have several lines starting with `PARAM: ` or something like `PARAM+:` which appends/overrides the previous line.

**Comment by sim642** [comment]
> We could just have several lines starting with `PARAM:`

Indeed, I've thought about this many times. Theoretically it should be an easy change: just repeat the same test file with each `PARAM` given. Where is gets complicated is how to report the results of such multi-param tests: if one of the configurations passes but another fails, then it should be clear which ones fail (not that just some failed).
Also it complicates the `regtest.sh` script, which then must have a way to choose the configuration you want to run.

What's worse is that it would be desirable to specify expected results differently for the different configurations. Otherwise one must set them based on the most imprecise configuration and use `// TODO` in places where some configuration could be more precise.

**Comment by vogler** [comment]
Good points. I guess it mostly makes sense for tests where the expected results are the same (e.g. groups of solvers).
Or we come up with some extra syntax to reference the configuration -- probably better to name them, just the index might be too cryptic.

For reporting we could diff the param lines to identify which config failed (and sth like: 1 out of 2 configurations failed: ...).
